app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -491,17 +483,73 @@ else if (possiblePoke[1] < 2) {
     /**
      * @return the likely pokemon number against the char sequence as well as the similarity
      */
-    private int[] getPossiblePokemon(CharSequence rhs) {
+    private int[] getPossiblePokemon(CharSequence poketext, CharSequence candytext) {
         int pokeNumber = 0;
         int bestMatch = 100;
-        for (int i = 0; i < pokeCalculator.pokedex.size(); i++) {
-            int similarity = pokeCalculator.get(i).getDistance(rhs);
-            if (similarity < bestMatch) {
-                pokeNumber = i;
-                bestMatch = similarity;
+
+        for (Pokemon p: pokeCalculator.pokedex) {
+            int dist = p.getDistance(poketext);
+            if (dist < bestMatch) {
+                pokeNumber = p.number;
+                bestMatch = dist;
+            }
+
+            /* We found a perfect match. We are done. */
+            if (bestMatch == 0) {
+                int[] result = {pokeNumber, bestMatch};
+                return result;
+            }
+        }
```

Comment:
You need to consider the case where neither are OCRed correctly. In that case, I'd want to go by the closest match. This loop is needed in my opinion.The loop itself has nothing to do with performance.